flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala

```diff
@@ -92,6 +92,17 @@ object ScalarFunctions {
     sb.toString
   }
 
+  /**
+    * Returns the string str with all non-overlapping occurrences
+    * of search replaced with replacement.
+    */
+  def replace(str: String, search: String, replacement: String): String = {
```

Comment:
The `replace()` method has been provided in `org.apache.calcite.runtime.SqlFunctions`.

Reply:
The same reason like the above. The calcite builtin `replace` can not handle null parameters, which may throw NPE. 
The builtin `replace` is
```
public static String replace(String s, String search, String replacement) {
    return s.replace(search, replacement);
  }
```
and in java.lang.String#replace it is
```
public String replace(CharSequence target, CharSequence replacement) {
        return Pattern.compile(target.toString(), Pattern.LITERAL).matcher(
                this).replaceAll(Matcher.quoteReplacement(replacement.toString()));
    }
```
Here it calls `toString` method, that will cause Exception when any of the thress params is `null`.